### DISCUSSION TITLE
CheckboxList with static options not saving - where to begin to debug?

### DISCUSSION THREAD

**Discussion by pekka** [filamentphp/filament#6881] [Help] [answered]
I have two `CheckboxList`s in a Filament resource with many fields. 

All the fields are saving fine, except them. After clicking "save", I get confirmation that saving worked, and see the checkboxes ticked. 

However, when I do a reload on the entire page, the checkboxes disappear, and the database seems to never receive those values. 

A third `CheckboxList` in the same form that gets its values from a relationship works just fine. 

They get their values from a static array via `->options()`. I've simplified it: 

```
$item->options(["1", "2", "3"]);
```

- The underlying database fields are JSON (I've tried TEXT, no change)

- They are defined in the model's `casts` property:
   
``` 
'allowed_booking_types' => 'array',
 'allowed_settlement_types' => 'array', 
```

Where would I begin to debug this? Is there a "before save" event I can plug into to see what is going wrong? 

Any pointers much appreciated.

**Comment by pboivin** [comment, marked as answer]
Did you also add the field to your model's `$fillable`?

**Reply by pekka** [reply]
`Facepalm emoji.jpg`

Of course. 🤦 Thank you.
